algosdk/abi/contract.py

```diff
@@ -63,6 +63,26 @@ def undictify(d: dict) -> "Contract":
             name=name, desc=desc, networks=networks, methods=method_list
         )
 
+    def get_method_by_name(self, name: str) -> Method:
+        methods_filtered = list(
+            filter(lambda method: method.name == name, self.methods)
+        )
+
+        if len(methods_filtered) > 1:
+            raise KeyError(
+                "Found {} methods with the same name {}".format(
+                    len(methods_filtered),
+                    ",".join(
+                        [method.get_signature() for method in methods_filtered]
+                    ),
+                )
+            )
```

Comment:
so this does not allow for contract with two methods of same name, but type-signatures are different? I mean `add1(uint32)uint32` and `add1(uint16)uint16`?

Do you think it would be more helpful if we want to return a list of `methods` or method-signatures, rather than just one method?

Reply:
ok, that makes sense